src/Raven.Client/Documents/AI/DatabaseAiAgents.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations.AI.Agents;
+using Raven.Client.Util;
+
+namespace Raven.Client.Documents.AI;
+
+/// <summary>
+/// Manages AI agents and chat interactions in a specific RavenDB database.
+/// </summary>
+public class DatabaseAiAgents
+{
+    private string _databaseName;
+    private IDocumentStore _store;
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="DatabaseAiAgents"/> for a given document store and optional database name.
+    /// </summary>
+    /// <param name="store">The RavenDB document store.</param>
+    /// <param name="databaseName">The name of the database. If null, uses the default database from the store.</param>
+    public DatabaseAiAgents(IDocumentStore store, string databaseName = null)
+    {
+        _databaseName = databaseName ?? store.Database;
+        _store = store;
+    }
+
+    /// <summary>
+    /// Returns a <see cref="DatabaseAiAgents"/> for a different database.
+    /// </summary>
+    /// <param name="databaseName">The name of the target database.</param>
+    /// <returns>A new or existing <see cref="DatabaseAiAgents"/> instance.</returns>
+    public DatabaseAiAgents ForDatabase(string databaseName)
+    {
+        if (string.Equals(_databaseName, databaseName, StringComparison.OrdinalIgnoreCase))
+            return this;
+
+        return new DatabaseAiAgents(_store, databaseName);
+    }
+
+    /// <summary>
+    /// Asynchronously creates or updates an AI agent configuration (with the given schema) on the database.
+    /// </summary>
+    /// <typeparam name="TSchema">The schema type the AI agent should use.</typeparam>
+    /// <param name="agentName">The unique name of the AI agent for create an agent, or name of an existed agent for update an agent.</param>
+    /// <param name="configuration">The configuration to assign to the agent.</param>
+    /// <returns>The result of the creation or update operation.</returns>
+    public async Task<AiAgentConfigurationResult> CreateAgentAsync<TSchema>(string agentName, AiAgentConfiguration configuration, CancellationToken token = default) where TSchema : new()
+    {
+        return await _store.Maintenance.ForDatabase(_databaseName).SendAsync(new AddOrUpdateAiAgentOperation<TSchema>(agentName, configuration), token).ConfigureAwait(false);
```

Comment:
let's do it the same way as for `TimeSeriesOperations`, when the executor is created once in the ctor, so no need to do `ForDatabase` every time